components/servo/lib.rs

```diff
@@ -1293,39 +1293,62 @@ enum UserAgent {
     iOS,
 }
 
-fn default_user_agent_string_for(agent: UserAgent) -> &'static str {
+fn get_servo_version() -> &'static str {
+    env!("CARGO_PKG_VERSION")
+}
+
+fn default_user_agent_string_for(agent: UserAgent) -> String {
+    let servo_version = get_servo_version();
+
     #[cfg(all(target_os = "linux", target_arch = "x86_64", not(target_env = "ohos")))]
-    const DESKTOP_UA_STRING: &str =
-        "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Servo/1.0 Firefox/111.0";
+    let desktop_ua_string = format!(
+        "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Servo/{} Firefox/111.0",
```

Comment:
I think this would look nicer inline here and elsewhere.

```suggestion
        "Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Servo/{servo_version} Firefox/111.0",
```